### DISCUSSION TITLE
Sound cards that work for stereo input (TRRS vs TRS)

### DISCUSSION THREAD

**Discussion by avit** [mcguirepr89/BirdNET-Pi#216] [Q&A] [answered]
I ordered [this UGREEN USB sound card](https://www.amazon.ca/gp/product/B08Y8CZB2S/ref=ppx_yo_dt_b_asin_title_o00_s01?ie=UTF8&psc=1) as it seems to have the same specs as the one recommended in the [suggested USB microphones](https://github.com/mcguirepr89/BirdNET-Pi/discussions/39) discussion.

It says "TRRS" and "2 in 1" and "mic" on the description. However, when I plug it in it doesn't show up as a usable sound input:

```
$ arecord -l
**** List of CAPTURE Hardware Devices ****

$ cat /proc/asound/cards
 0 [vc4hdmi        ]: vc4-hdmi - vc4-hdmi
                      vc4-hdmi
 1 [Audio          ]: USB-Audio - KT USB Audio
                      KTMicro KT USB Audio at usb-3f980000.usb-1, full speed
```

Alsamixer says "This sound device does not have any capture controls" when I select the KT USB Audio sound card.

Has anyone got this one working, or is there a reason it's not showing up? I'd like to confirm what I'm missing before I order another one that looks nearly identical; or maybe there's another one you can recommend.

**Comment by avit** [comment]
I am now suspecting the problem might be my cable. Since the audio device disappears when I disconnect it, there must be some automatic detection there when it's plugged in.

Reading [some more](https://github.com/mcguirepr89/BirdNET-Pi/discussions/39#discussioncomment-1525783), I had wired a pair of condenser mics onto what turns out to be an old TRS stereo plug, thinking that I might be able to record in stereo.

[This article](https://github.com/DD4WH/SASS/wiki/Stereo--(Mono)-recording-low-noise-low-cost-system) suggests that some UGREEN USB adapters work with stereo input. That doesn't seem to be the case with the one I bought, or else it's the TRS plug...

**Reply by CaiusX** [reply]
Hi Avit

It's likely the plug, grab a stereo trrs from an old stereo headphone & try wire that up.

**Comment by nilspupils** [comment]
I have exactly the same soundcard and it works fine. It uses TRRS so will only be mono for the microphone. Be aware there are two different TRRS standards, so not all microphones/headsets will work. This one has the microphone input on "S" (sleeve) and ground on the second ring! Only electret type microphones will work (not dynamic). 
It should be autodetected if a mic is present.

**Reply by avit** [reply]
Thanks for confirming. Does second ring mean the one just above the sleeve? Or the next one by the tip? I'm trying to understand if a TRS jack can work in the TRRS adapter, with a mono mic only, or if there needs to be a separation with the extra ring.

I might have to find a TRRS cable then.

**Reply by nilspupils** [reply]
Second ring is the one next to the sleeve. You cannot use a TRS plug since it would put the microphone signal to ground. There are adapters that split TRRS into TRS (Headphone L+R) and TRS (Microphone) but in this scenario still would leave you with just one channel.
For stereo recording you will need an interface with a seperate mic input (and stereo mic input specified).

**Comment by DD4WH** [comment, marked as answer]
I have modified the post here https://github.com/mcguirepr89/BirdNET-Pi/discussions/39#discussion-3631548
which might probably answer your question regarding TRS / TRRS and which sound cards support Stereo input.

**Reply by avit** [reply]
Very helpful. That makes sense now. Thanks!
